### PR TITLE
Formatting of the getSection API command

### PR DESCRIPTION
**Pull request by lamrowena**
Addresses issue #47. Updated the output to be an array rather than an object and included example output.

### CODE REVIEW

File: Core/CMP API Specification.md [hunk 1]
```diff
@@ -458,8 +458,31 @@ For example, client can ask the CMP to get the IAB TCF CA v1.0 TCData:
...
+ {
+  SegmentType:3, 
+  PubPurposesExpressConsent : [1,2,3,4,5],
+  PubPurposesImpliedConsent  : [6,7,8,9],
+  ...
+  } 
+ ]
+```
+
 A call to ` __gpp('getSection', null, "tcfeuv2");` will return `null`.
```
Comment (janwinkler) [inline Core/CMP API Specification.md:486]: lets not include this text. that is a policy decision and not a tech decision.

**Review by janwinkler — COMMENTED** [review]
getSection:
- its actually now an array of objects and not an object anymore (in description and the table with callback function (data: ....)
- lets make what "data" is more specific and not just as an example how the output looks

Suggested text:
getSection
The `getSection` command can be used to receive the (parsed) representation of a section of a certain specification. The callback shall be called with the (parsed) representation as the argument for the `data` parameter. The parsed representation of a section is an array of objects, where each object represents one segment of this section in the order that is given by the section specification. For example, the `data` parameter for the TCF Canada will be an array with one or two objects (Core segment plus optional publisher purposes segment). Each object is composed of the fields given by the section specification.
The `data` parameter may be `null` for sections that don't allow accessing the section data object outside an event handler.  It may also be `null` when the CMP is not yet loaded.

**Comment by janwinkler** [comment]
Additional necessary change needed: getField "parameter" value descrition and example.

Proposal:
...
parameter - string - API Prefix string + "." (dot) + [Section index* + "." (dot)] + fieldname
return - ..... (stays as it is) ...

The section index is the index of a section within an segment, starting with 0 (zero) for the first section. The section index is optional, if not used, the first section is assumed.

For example, a client can ask the CMP to get the last updated field from the IAB TCF **Canada v1.0** TCData.

var s = __gpp('getField',null, "tcf**ca**.LastUpdated");
or
var s = __gpp('getField',null, "tcfca.0.LastUpdated");

In order to get the Publisher Purposes Express Consent information from IAB TCF Canada the command would be used as:
var s = __gpp('getField',null, "tcfca.1.PubPurposesExpressConsent");

File: Core/CMP API Specification.md (hunk 1 above)
Comment (HeinzBaumann) [reply Core/CMP API Specification.md:487]: I can remove it.

File: Core/CMP API Specification.md [hunk 2]
```diff
@@ -458,8 +459,31 @@ For example, client can ask the CMP to get the IAB TCF CA v1.0 TCData:
 
 
 ```javascript
-__gpp('getSection', myFunction, "tcfcav1");
+__gpp('getSection', myFunction, "tcfca");
```
Comment (patmmccann) [inline Core/CMP API Specification.md:462]: pingReturn gives a list of applicable section numbers. How is an api caller supposed to know which section string to call for some number? Should they only listen for sectionChange?
image

**Review by patmmccann — APPROVED** [review]
Lgtm, my other comments can be addressed in ongoing efforts

File: Core/CMP API Specification.md [hunk 3]
```diff
@@ -415,17 +415,18 @@ The `hasSection` command can be used to detect if the CMP has generated a sectio
...
-__gpp('hasSection', myFunction, "tcfeuv2");
+__gpp('hasSection', myFunction, "tcfcav1");
 ```
 
 ______
 #### `getSection` <a name="getsection"></a>
 
-The `getSection` command can be used to receive the (parsed) object representation of a section of a certain specification. 
-The callback shall be called with the (parsed) object representation as the argument for the `data` parameter. The `data` parameter may be `null` for sections that don't allow accessing the section data object outside an event handler.  It may also be `null` when the CMP is not yet loaded.
+The `getSection` command can be used to receive the (parsed) representation of a section of a certain specification. The callback shall be called with the (parsed) representation as the argument for the data parameter. The parsed representation of a section is an array of objects, where each object represents one segment of this section in the order that is given by the section specification. For example, the data parameter for the TCF Canada will be an array with one or two objects (Core segment plus optional publisher purposes segment). Each object is composed of the fields defined by the section specification.
```
Comment (janwinkler) [inline Core/CMP API Specification.md:427]: segment => subsection

File: Core/CMP API Specification.md [hunk 4]
```diff
@@ -458,9 +459,32 @@ For example, client can ask the CMP to get the IAB TCF CA v1.0 TCData:
...
 
 ```javascript
-__gpp('getSection', myFunction, "tcfcav1");
+__gpp('getSection', myFunction, "tcfcav1"); 
 ```
-A call to ` __gpp('getSection', null, "tcfeuv2");` will return `null`.
+Example value of data passed to the callback: 
+```javascript
+[
+ /* Core segment */
```
Comment (janwinkler) [inline Core/CMP API Specification.md:467]: Core subsection

File: Core/CMP API Specification.md [hunk 5]
```diff
@@ -458,9 +459,32 @@ For example, client can ask the CMP to get the IAB TCF CA v1.0 TCData:
...
+ {
+  Version:1, 
+  Created: Date (Thu Apr 13 2023 18:07:12 GMT+0200),
+  LastUpdated: Date (Thu Apr 13 2023 18:07:12 GMT+0200),
+  CmpId: 31, 
+  CmpVersion: 123,
+  ConsentScreen: 5,
+  ...
+  }, 
+  /* Publisher Purposes Segment (optional) */
```
Comment (janwinkler) [inline Core/CMP API Specification.md:477]: Publisher purposes subsection (optional)

File: Core/CMP API Specification.md [hunk 6]
```diff
@@ -458,9 +459,32 @@ For example, client can ask the CMP to get the IAB TCF CA v1.0 TCData:
...
+ {
+  SegmentType:3,
```
Comment (janwinkler) [inline Core/CMP API Specification.md:479]: rename this field to subsectiontype?

File: Core/CMP API Specification.md (hunk 3 above)
Comment (lamrowena) [reply Core/CMP API Specification.md]: done

File: Core/CMP API Specification.md (hunk 4 above)
Comment (lamrowena) [reply Core/CMP API Specification.md]: done

File: Core/CMP API Specification.md (hunk 6 above)
Comment (lamrowena) [reply Core/CMP API Specification.md]: done

**Pull request merged**